test/Driver/sanitizers.swift

```diff
@@ -20,6 +26,24 @@
 // RUN: not %swiftc_driver -resource-dir %S/Inputs/fake-resource-dir/lib/swift/ -driver-print-jobs -sanitize=thread -target x86_64-unknown-windows-msvc %s 2>&1 | %FileCheck -check-prefix=TSAN_WINDOWS %s
 // RUN: %swiftc_driver -resource-dir %S/Inputs/fake-resource-dir/lib/swift/ -driver-print-jobs -sanitize=thread -target x86_64-unknown-linux-gnu %s 2>&1 | %FileCheck -check-prefix=TSAN_LINUX %s
 
+/*
+ * Undefined Behavior Sanitizer Tests  (ubsan)
+ */
+
+// RUN: %swiftc_driver -resource-dir %S/Inputs/fake-resource-dir/lib/swift/ -driver-print-jobs -sanitize=undefined -target x86_64-apple-macosx10.9 %s | %FileCheck -check-prefix=UBSAN -check-prefix=UBSAN_OSX %s
+// RUN: not %swiftc_driver -resource-dir %S/Inputs/fake-resource-dir/lib/swift/ -driver-print-jobs -sanitize=undefined -target x86-apple-macosx10.9 %s 2>&1 | %FileCheck -check-prefix=UBSAN_OSX_32 %s
+// RUN: not %swiftc_driver -resource-dir %S/Inputs/fake-resource-dir/lib/swift/ -driver-print-jobs -sanitize=undefined -target x86_64-apple-ios7.1 %s 2>&1 | %FileCheck -check-prefix=UBSAN_IOSSIM %s
```

Comment:
Hm, this doesn't seem right. I'm pretty sure we support UBSan on all Apple platforms. @cheshire?

Reply:
I think you’re right - I was largely basing these off the existing tsan tests and didn’t see the “not” prefix

Reply:
Also one thing I’ve been trying to find an answer for is if ubsan is supported on Windows. If so I’ll update the windows toolchain as well

Reply:
Looks like not yet. Even if we're wrong about that, we can always add it later.

Reply:
Ok I've omitted it for now. Updated the tests, let me know if this looks right